includes/documents/abstract-wcpdf-order-document.php

```diff
@@ -103,6 +134,12 @@ abstract class Order_Document {
 	 * @param  int|object|WC_Order $order Order to init.
 	 */
 	public function __construct( $order = 0 ) {
+		// semaphore
+		$this->lock_name    = "wpo_wcpdf_{$this->slug}_semaphore_lock";
+		$this->lock_time    = apply_filters( 'wpo_wcpdf_document_semaphore_lock_time', 60 );
```

Comment:
Maybe it's good to also send the document slug/type as a parameter to this filter?